### DISCUSSION TITLE
ATN

### DISCUSSION THREAD

**Discussion by Ahmed-Emad10** [antlr/antlr4#4352] [General]
Hello there,
Hope you are well. I just wanted to ask what is ATN stands for?
Also what is used for?
Thanks in advance.

**Comment by ericvergnaud** [comment]
Augmented Transition NetworkFor antlr, it describes, starting from a given sequence of tokens compliant with a grammar, which other valid sequences are reachable when adding the next token discovered by the lexer. Envoyé de mon iPhoneLe 9 juil. 2023 à 16:35, Ahmed Emad ***@***.***> a écrit :﻿
Hello there,
Hope you are well. I just wanted to ask what is ATN stands for?
Also what is used for?
Thanks in advance.
